src/components/shared/Map/Legend.tsx

```diff
@@ -0,0 +1,56 @@
+import { COLORS } from '@/constants/index';
+
+export default function Legend() {
+  const pollutionLevels: string[] = [
+    'Bardzo dobra',
+    'Dobra',
+    'Umiarkowana',
+    'Dostateczna',
+    'Zła',
+    'Bardzo zła',
+    'Niebezpieczna'
+  ];
+  const legendData: Array<{
+    id: number;
+    color: string | undefined;
+    description: string | undefined;
+  }> = [];
+
+  for (let i = 0; i < pollutionLevels.length; i += 1) {
+    const levelData = {
+      id: i,
+      color: COLORS.pollutionScale[i],
+      description: pollutionLevels[i]
+    };
+    legendData.push(levelData);
+  }
+
+  return (
+    <div className='absolute top-40 right-4 bg-black bg-opacity-50 rounded z-10'>
```

Comment:
```suggestion
    <div className='absolute top-4 right-4 bg-black bg-opacity-50 rounded z-10'>
```
Maybe make distances the same (from top and right screen side). It looks a bit strange when a distance from the top is so large.

Reply:
alright